client/internal/engine.go

```diff
@@ -462,8 +463,8 @@ func (e *Engine) modifyPeers(peersUpdate []*mgmProto.RemotePeerConfig) error {
 	var modified []*mgmProto.RemotePeerConfig
 	for _, p := range peersUpdate {
 		peerPubKey := p.GetWgPubKey()
-		if peerConn, ok := e.peerConns[peerPubKey]; ok {
-			if peerConn.WgConfig().AllowedIps != strings.Join(p.AllowedIps, ",") {
+		if allowedIP, ok := e.peerStore.AllowedIP(peerPubKey); ok {
+			if allowedIP.String() != strings.Join(p.AllowedIps, ",") {
```

Comment:
I think this is different than before. `allowedIP` is a `net.IP`, so always a single one, while the one from wgconfig could be multiple